### ISSUE TITLE
dD Triangulation Finite_facet_iterator does not compile

### ISSUE THREAD

**Issue by Abacn** [CGAL/cgal#3274]
_Please use the following template to help us solving your issue._

## Issue Details

Hi everyone. I am trying to traverse the finite facets of dD triangulation. I found the iteration method such that finite_facets_begin() doesn't compile, but finite_full_cell_begin() or finite_vertices_begin() works well. Here is a working example

## Source Code

```cpp
#include <CGAL/Epick_d.h>
#include <CGAL/point_generators_d.h>
#include <CGAL/Triangulation.h>
#include <CGAL/algorithm.h>

#include <iostream>
#include <iterator>
#include <vector>

#define DIM 4
typedef CGAL::Epick_d<CGAL::Dimension_tag<DIM> > DIM_tag;
typedef CGAL::Triangulation< DIM_tag > Triangulation;

int main()
{
  const int N = 100; // we will insert 100 points
  CGAL::Random_points_in_cube_d<Triangulation::Point> rand_it(DIM, 1.0);
  std::vector<Triangulation::Point> points;
  CGAL::cpp11::copy_n(rand_it, N, std::back_inserter(points));
  Triangulation t(DIM);                      // create triangulation
  t.insert(points.begin(), points.end());  // compute triangulation
  // Finite_full_cell_iterator or Finite_vertex_iterator both work
  int i = 0;
  for( auto cit = t.finite_full_cells_begin();
      cit != t.finite_full_cells_end(); ++cit )
  {
    ++i;
  }
  std::cout << "There are " << i << " finite full cells."<< std::endl;
  
  // Finite_facet_iterator compile failed
  int j = 0;
  for( auto fit = t.finite_facets_begin() ;
      fit != t.finite_facets_end(); ++fit )
  {
    ++j;
  }
  std::cout << "There are " << j << " finite facets."<< std::endl;
  return 0;
}
```

## Guess

I look into the header file Triangulation.h, and find that

Line 461:
```cpp
    Facet_iterator facets_begin() { return tds().facets_begin(); }
    Facet_iterator facets_end() { return tds().facets_end(); }
    Facet_iterator finite_facets_begin()
    { return Finite_facet_iterator(Finiteness_predicate(*this), facets_begin(), facets_end()); }
    Facet_iterator finite_facets_end()
    { return Finite_facet_iterator(Finiteness_predicate(*this), facets_end(), facets_end()); }
```
It seems that the return type of last two declarations should be  Finite_facet_iterator instead. But simply changing these doesn't resolve the compilation error. Maybe need to recompile library file?

## Environment

* Operating system (Max, 32/64 bits):
* Compiler: LLVM 9.0
* Release or debug mode: debug
* Specific flags used (if any):
* CGAL version: 4.12
* Boost version: 1.67.0_1
* Other libraries versions if used (Eigen: stable 3.3.5):

**Comment by MaelRL** [comment]
Thanks for the bug report, `Facet_iterator` was indeed the wrong type.

Please see https://github.com/CGAL/cgal/pull/3275 for the fix.

**Comment by lrineau** [comment]
Fixed by #3275.
